SoftLayer/managers/hardware.py

```diff
@@ -257,6 +257,28 @@ def rescue(self, hardware_id):
         """
         return self.hardware.bootToRescueLayer(id=hardware_id)
 
+    def trunk(self, hardware_id, public, vlan_id):
```

Comment:
If we're already requiring the vlan_id to trunk, can't we  look up if this is a public or private vlan and trunk to the correct network interface using that information?